test/orogen_test_helpers.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+module IODriversBase
+    # Common helpers for the rather complex setup of iodrivers_base::Task tests
+    #
+    # It relies on the test cases to deploy a task-under-test and store it in the
+    # @task instance variable
+    module OroGenTestHelpers
+        # The task under test
+        attr_reader :task
+
+        def setup
+            super
+
+            @__iodrivers_base_sockets = []
+        end
+
+        def teardown
+            super
+
+            @__iodrivers_base_sockets.each(&:close)
+        end
+
+        # Configure @task to use a file descriptor
+        #
+        # Internally, it creates a UDP socket and set the io_port property to connect to it
+        def setup_iodrivers_base_with_fd
```

Comment:
Would you consider receiving the task as an argument and storing it in a "hidden" member (`@__task` ?) to use in the composition?

Reply:
Since the composition and FD are mutually exclusive, I just changed the helpers to be given the task as argument. The only state variable are the sockets that need to be closed on teardown.